<launch>

    <arg name="exposure" default="100" />
    <arg name="gain" default="100" />
    <arg name="FPS" default="50" />
    <arg name="resolution_width" default="320" />
    <arg name="resolution_height" default="240" />
    <arg name="publish_raw" default="1" />


    <node pkg="duo3d_ros" name="duo_node" type="duo_node" required="true" output="screen">
		<!-- Supported Device Names: [Mini, Mini-lx] -->
		<param name="device_name" 		value="Mini-lx" />
		<param name="device_serial_number" 	value="N-A" />

    	<param name="frame_id"   value="duo3d_camera" />
		<!-- 	Choose anywhere from 0.1-3000 Frames Per Second (FPS) for the DUO to publish at. Please only
		use up to one decimal place ONLY, if you need to. This does not require a specific rate, like 
		the duo_dense3d.launch file requires. 

		 56 FPS @ 752x480 ->  49
		 62 FPS @ 640x480
		123 FPS @ 640x240
		240 FPS @ 640x120
		 93 FPS @ 320x480
		184 FPS @ 320x240 -> 171
		360 FPS @ 320x120

		-->	
		<param name="FPS"   value="$(arg FPS)" />

		<param name="resolution_width"	value="$(arg resolution_width)" />
		<param name="resolution_height"	value="$(arg resolution_height)" />


		<param name="exposure"      value="$(arg exposure)"  />
		<param name="gain"          value="$(arg gain)"  /> 
		<param name="led_lighting"	value="0"  />

	    <param name="use_DUO_imu"   	value="true" />
	    <param name="use_DUO_LEDs"   	value="true" />

	    <param name="VerticalFlip"   	value="false" />
	    <param name="HorizontalFlip"   	value="false" />
	    <param name="CameraSwap"     	value="true" />
	    
	    <param name="publish_raw" type="bool" value="$(arg publish_raw)" />

	    <!-- NOTE: to set the calibration manually you'll need to have the "ost.txt" file from the ros calibration. then you convert (use http://wiki.ros.org/camera_calibration_parsers) it to two separate files and put it into the calibration folder -->

	    <param name="camera_info_left" value="file://$(find duo3d_ros)/calibration/duo3d_left.yaml" />
	    <param name="camera_info_right" value="file://$(find duo3d_ros)/calibration/duo3d_right.yaml" />
    </node>

  <include file="$(find duo3d_ros)/launch/auto_exposure.launch">
  </include>

</launch>
